tendrl/commons/objects/__init__.py

```diff
@@ -83,11 +83,13 @@ def load_definition(self):
 
     @thread_safe
     def save(self, update=True, ttl=None):
+        hash_key_changed = True
         if "Message" not in self.__class__.__name__:
             # If local object.hash is equal to
             # central_store object.hash, return
             if self.hash_compare_with_central_store(ttl=ttl):
-                return
+                # No change in hashkey
+                hash_key_changed = False
```

Comment:
How this is different from old code. Earlier if there was no change in hash we were returning. Now also with new code you don't perform anything if there is no change in hash.

Reply:
In old code watcher logic are not called it won't execute further after step 90, but if you see the watcher steps are written below this. 

So this flag will won't allow writing "data" field in etcd. But watcher is updated.

Reply:
oh ok. ack